mitosheet/mitosheet/tests/step_performers/test_replace.py

```diff
@@ -350,13 +350,233 @@
 
 
+REPLACE_SELECTED_COLUMNS = [
+    # Tests with boolean columns
+    (
+        [
+            pd.DataFrame({
+                'test1': [1, 2, 3],
+                'test2': [1.0, 2.0, 3.0], 
+                'test3': [True, False, True], 
+                'test4': ["string", "with spaces", "and/!other@characters3"], 
+                'test5': pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+        0,
+        [ "test4" ],
+        "3", 
+        "4", 
+        [
+            pd.DataFrame({
+                'test1': [1, 2, 3],
+                'test2': [1.0, 2.0, 3.0], 
+                'test3': [True, False, True], 
+                'test4': ["string", "with spaces", "and/!other@characters4"], 
+                'test5': pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+    ),
+    # Replace boolean and non-boolean columns
+    (
+        [
+            pd.DataFrame({
+                'test1': [1, 2, 3],
+                'test2': [1.0, 2.0, 3.0], 
+                'test3': [True, False, True], 
+                'test4': ["string", "with spaces", "and/!other@characters3"], 
+                'test5': pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+        0,
+        [ "test3", "test4" ],
+        "a", 
+        "b", 
+        [
+            pd.DataFrame({
+                'test1': [1, 2, 3],
+                'test2': [1.0, 2.0, 3.0], 
+                'test3': [True, False, True], 
+                'test4': ["string", "with spbces", "bnd/!other@chbrbcters3"], 
+                'test5': pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+    ),
+
+    # Tests with non-string column headers
+    (
+        [
+            pd.DataFrame({
+                1: [1, 2, 3],
+                2: [1.0, 2.0, 3.0], 
+                3: [True, False, True], 
+                4: ["string", "with spaces", "and/!other@characters3"], 
+                5: pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+        0,
+        [ "4" ],
+        "a", 
+        "b", 
+        [
+            pd.DataFrame({
+                1: [1, 2, 3],
+                2: [1.0, 2.0, 3.0], 
+                3: [True, False, True], 
+                4: ["string", "with spbces", "bnd/!other@chbrbcters3"], 
+                5: pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+    ),
+    (
+        [
+            pd.DataFrame({
+                1: [1, 2, 3],
+                2: [1.0, 2.0, 3.0], 
+                3: [True, False, True], 
+                4: ["string", "with spaces", "and/!other@characters3"], 
+                5: pd.to_datetime(['12-22-1997', '12-23-1997', '12-24-1997']), 
+            })
+        ],
+        0,
+        [ "4" ],
+        "4", 
+        "6", 
+        [
+            pd.DataFrame({
+                1: [1, 2, 3],
```

Comment:
Check out these tests -- all of the dataframes have column headers of the same type. Think about the dataframes you've seen in the wild -- rarely do they have only numbers or only strings as headers.

Let's add some tests for this, which I think would catch the dtype bug with the index.

Try and think about each critical line of code you write - what test can you add that will stress test if it's correct?